redaxo/src/core/lib/login/login.php

```diff
@@ -239,8 +282,19 @@ public function checkLogin()
                 // message schreiben und falls falsch auf error verweisen
 
                 $ok = true;
-                $sessionStartStamp = (int) $this->getSessionVar('STAMP');
-                if (($sessionStartStamp + $this->sessionDuration) < time()) {
+
+                // check session max age
+                $sessionStartStamp = rex_request::session(self::START_TIME, 'int');
```

Comment:
Das hast du vermutlich bewusst hier so gemacht, dass du nicht den Session-Bereich des Logins nutzt, sondern die komplette Session der REDAXO-Instanz, oder?

Ich bin hier noch am überlegen, ob ich das sinnvoll so finde.

Was wäre zum Beispiel mit diesem Fall:
- Frontend nutzt auch Session, auch im nicht eingeloggten Bereich. Also Session wird immer gestartet.
- An Tag X rufe ich Frontend auf, Session wird gestartet, somit auch START_TIME gesetzt
- Ab dann bin ich regelmäßig auf der Seite, die Session wird also über lange Zeit erhalten
- nach fast 4 Wochen logge ich mich dann im Backend ein
- dann ist meine Backendsession dann plötzlich nur kurz gültig, oder? weil start_time ja vor fast 4 Wochen

Also ich bin wie gesagt noch am überlegen. Vielleicht ist der Fall auch zu konstruiert.

Aber anders gefragt: Was spricht für dich dagegen, es über `$this->getSessionVar` zu machen, also spezifisch jeweils nur für den login?

Reply:
ich denke der realistischere fall wäre dass jemand mehrere redaxo instanzen auf einem server hat und man in instanz B ausgeloggt wird, obwohl man instanz A seit langem nicht genutzt hat.

wie auch immer: es macht sinn `getSessionVar`  zu verwenden.. ich hatte beide Fälle nicht bedacht, guter Hinweis.